### ISSUE TITLE
About PPG implementations

### ISSUE THREAD

**Issue by RajS999** [09tangriro/ppg#1]
Hi, I came across your repos while learning PPG. I found following three PPG implementations:

1. https://github.com/09tangriro/ppg/ (this / current one)
2. https://github.com/09tangriro/brain-PUBLIC-VERSION/tree/dev/brain/ppg
3. https://github.com/09tangriro/stable-baselines3-contrib/blob/sandbox/ppg/sb3_contrib/ppg/ppg.py

Want to know which one of these is correct and if you have done any (correctness / performance) evaluation for them. How well they compare with openai implementation?

I was able to run the code from `brain` (2nd one) repo, though I havent analysed the output yet (just quick run). Didnt tried 1st repo. Got following errors while trying out 3rd one:

 - `'PPG' object has no attribute 'policy_epochs'` at [line 203](https://github.com/09tangriro/stable-baselines3-contrib/blob/ad8bff5e76b5b828a174f5ecb0e29eec293460a6/sb3_contrib/ppg/ppg.py#L203). You seem to have forgot to initialize it 
 - `name 'policy_loss_1' is not defined` at [line 231](https://github.com/09tangriro/stable-baselines3-contrib/blob/ad8bff5e76b5b828a174f5ecb0e29eec293460a6/sb3_contrib/ppg/ppg.py#L231). Seems that they should be named `surrogate_loss_1` and `surrogate_loss_2`?
 - 'ActorCritic' object has no attribute 'actor'` at [line 257](https://github.com/09tangriro/stable-baselines3-contrib/blob/ad8bff5e76b5b828a174f5ecb0e29eec293460a6/sb3_contrib/ppg/ppg.py#L257). I believe same error will be there for `critic`.

Or am simply doing something wrong due to which getting above errors?

PS: sorry, this question might not be suited for issues, may be I could have raised discussion.

**Comment by 0xprofessooor** [comment]
Hey! Yeah sorry about the mess on the repos 😅

This one is the working version (although I haven't tested it against PPO tbh, just validated that it passes some environments). The second one, 'brain' is functional, but it doesn't really work. I ended up abandoning it to work on the third one where I'm adding the algo to Stable Baselines 3 Contrib library. Problem is, there's a dependency on some work I did recently on the Stable Baselines 3 repo so I need to wait for the next version of that to come out before I can continue with it.

Basically, this is the one that works properly, check out the blog posts attached in the README for further explanations!!

(I'll also attach notes to the other repos explaining they're on pause or abandoned)
